modyn/evaluator/internal/pytorch_evaluator.py

```diff
@@ -0,0 +1,177 @@
+import io
+import logging
+import multiprocessing as mp
+import os
+import pathlib
+import queue
+import traceback
+from inspect import isfunction
+from typing import Any, Callable, Optional, Union
+
+import torch
+from modyn.evaluator.internal.dataset.evaluation_dataset import EvaluationDataset
+from modyn.evaluator.internal.utils import EvaluationInfo, EvaluatorMessages
+
+EVALUATION_TRANSFORMER_FUNC_NAME = "evaluation_transformer"
+LABEL_TRANSFORMER_FUNC_NAME = "label_transformer_function"
+
+
+class PytorchEvaluator:
+    def __init__(
+        self,
+        evaluation_info: EvaluationInfo,
+        status_query_queue: mp.Queue,
+        status_response_queue: mp.Queue,
+        logger: logging.Logger,
+    ) -> None:
+        self.logger = logger
+        self._evaluation_id = evaluation_info.evaluation_id
+
+        self._model = evaluation_info.model_handler(
+            evaluation_info.model_configuration_dict, evaluation_info.device, evaluation_info.amp
+        )
+        self._load_state(evaluation_info.model_path)
+
+        # setup dataloaders
+        self._info("Setting up data loaders.")
+        self._dataloader = self._prepare_dataloader(evaluation_info)
+
+        self._mod_dict: dict[str, Any] = {}
+        self._evaluation_transformer = self._setup_transformer_function(
+            evaluation_info.evaluation_transformer, EVALUATION_TRANSFORMER_FUNC_NAME
+        )
+
+        self._metrics = evaluation_info.metrics
+
+        self._label_tranformer_function = self._setup_transformer_function(
+            evaluation_info.label_transformer, LABEL_TRANSFORMER_FUNC_NAME
+        )
```

Comment:
in general I really like this approach, but as said in another comment, if we introduce it, let's unify it in all places where we expect code in the pipeline dsecription

Reply:
I agree, done.